src/components/Charts/mutationsTree.js

```diff
@@ -0,0 +1,33 @@
+
+import React from 'react';
+import Container, {Title} from "./styles";
+
+class MutationsTree extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    createVariants = (variants, level) => {
+        return (variants.flatMap((variant) => [
+            `${'---'.repeat(level)}variant: ${variant.name} (${variant.mutations.map(mutation=>mutation.from+mutation.position+mutation.to).join(", ")})`,
+            ...this.createVariants(variant.subs, level+1)
+         ])).map(i => {
+            return <p>{i}</p>
+        });
+    }
+    
+    render() {
+        return (
+            <Container width="90%" ref="">
+                <Title>
+                    {"Mutations tree"}
+                </Title>
+                {this.createVariants(this.props.data, 0)}
```

Comment:
what about this? isn't it the simplest way how to achieve what we want, without using hooks, etc?

Reply:
sure, assuming this is performant enough (I have no idea whether the caching in previous version was because of performance or not).

Anyway, this looks good, let's just rename it to renderVariants